manim/config.py

```diff
@@ -1,283 +1,406 @@
-
-    return camera_config
+    # As before, make FILE_WRITER_CONFIG by pulling form the overall
+    config['FILE_WRITER_CONFIG'] = {key: config[key] for key in [
+        "WRITE_TO_MOVIE", "SAVE_LAST_FRAME", "SAVE_PNGS",
+        "SAVE_AS_GIF", "PNG_MODE", "MOVIE_FILE_EXTENSION",
+        "OUTPUT_FILE", "INPUT_FILE"]}
 
+    return config
 
-def initialize_directories(config):
-    dir_config = {}
-    dir_config["media_dir"] = config["media_dir"] or dirs.MEDIA_DIR
-    dir_config["video_dir"] = config["video_dir"] or dirs.VIDEO_DIR
 
-    if not (config["video_dir"] and config["tex_dir"]):
-        if config["media_dir"]:
-            if not os.path.isdir(dir_config["media_dir"]):
-                os.makedirs(dir_config["media_dir"])
-        if not os.path.isdir(dir_config["media_dir"]):
-            dir_config["media_dir"] = "./media"
+def _init_dirs(config):
+    if not (config["VIDEO_DIR"] and config["TEX_DIR"]):
+        if config["MEDIA_DIR"]:
+            if not os.path.isdir(config["MEDIA_DIR"]):
+                os.makedirs(config["MEDIA_DIR"])
+        if not os.path.isdir(config["MEDIA_DIR"]):
+            config["MEDIA_DIR"] = "./media"
         else:
             print(
-                f"Media will be written to {dir_config['media_dir'] + os.sep}. You can change "
-                "this behavior with the --media_dir flag, or by adjusting dirs.py.,"
+                f"Media will be written to {config['media_dir'] + os.sep}. You can change "
+                "this behavior with the --media_dir flag, or by adjusting manim.cfg."
             )
     else:
-        if config["media_dir"]:
+        if config["MEDIA_DIR"]:
             print(
                 "Ignoring --media_dir, since both --tex_dir and --video_dir were passed."
             )
 
-    dir_config["tex_dir"] = (config["tex_dir"]
-                             or dirs.TEX_DIR
-                             or os.path.join(dir_config["media_dir"], "Tex"))
-    dir_config["text_dir"] = (config["text_dir"]
-                              or dirs.TEXT_DIR
-                              or os.path.join(dir_config["media_dir"], "texts"))
+    # Make sure all folders exist
+    for folder in [config["VIDEO_DIR"], config["TEX_DIR"], config["TEXT_DIR"]]:
+        if not os.path.exists(folder):
+            os.makedirs(folder)
 
-    if not config["video_dir"] or dirs.VIDEO_DIR:
-        dir_config["video_dir"] = os.path.join(dir_config["media_dir"], "videos")
 
-    for folder in [dir_config["video_dir"], dir_config["tex_dir"], dir_config["text_dir"]]:
-        if folder != "" and not os.path.exists(folder):
-            os.makedirs(folder)
+config_filename = 'manim.cfg'
+
+foobar =  os.path.join(os.path.dirname(__file__), config_filename)
+print(foobar)
+config_files = [
+    foobar,
+    os.path.expanduser('~/.{}'.format(config_filename)),
+    os.path.join(os.getcwd(), config_filename),
+    ]
+
+prog = os.path.split(sys.argv[0])[-1]
+if prog in ['manim', 'manimcm']:
```

Comment:
The `__name__` is set to `manim.config`, even when called from the command line. `__main__` will only be "__main__" inside the `main.py` file.

Reply:
Yup, now I see why this was a problem. As a potential solution we could try setting a variable in `__main__` and checking here to see if it was set.

Reply:
Interesting. So in this case `config.py` would have to import `__main__.py` too, correct? If that's the case, then it sounds like it could get messy since `__main__.py` is also importing `config.py`....

Reply:
That's actually true, something like
```python
import config.IS_MAIN
if __name__ == "__main__":
    config.IS_MAIN = true
```
like I was thinking would cause config to initialize as soon as config was imported. I guess we could store `IS_MAIN` in a different module if it came to it, but that would be sort of awkward.

I'll leave it up to you since you said you've found a solution, but it's always an option should you need it.

Reply:
You can find my hacky solution in `_from_command_line()` in the latest commit.